### ISSUE TITLE
Epoch size

### ISSUE THREAD

**Issue by aleksandar** [davidsandberg/facenet#374]
Why we don't use whole training dataset in each epoch? There is example for casia to use 1000 iterations (epoch size) with batch size 90 and it is 1/5 of whole casia set per epoch.

I would like to say thanks @davidsandberg for this great project

**Comment by davidsandberg** [comment]
Hi,
The parameter `epoch_size`  should rather be named `evaluate_every_n_steps` or similar since it has nothing to do with the number of examples in the dataset. Hopefully this will be fixed in the future...

**Comment by aleksandar** [comment]
Hi,
Thanks a lot for this answer. Also there is another question regarding coverage of training set, there is a (small) chance to miss some classes during training because for every epoch we randomly choose `epoch_size * batch_size` number of images. If this number cover whole training set we'll eliminate this possibility to miss some classes. 

Again, thanks a lot for the project and the effort.

**Comment by davidsandberg** [comment]
For train_softmax.py that is not the case but I guess you are you referring to train_tripletloss.py?
In that case one epoch really consists of all triplets that are possible to generate from the dataset, and this will typically be a huge(!!) number and not possible to traverse during training. So there will anyway be some degree of random selection during training.

**Comment by aleksandar** [comment]
Actually, i am looking at the code in train_softmax.py. 
`index_queue = tf.train.range_input_producer(range_size, num_epochs=None,
                             shuffle=True, seed=None, capacity=32)`
Because this shuffle arg i was guessing it takes random images for epoch. Now i am thinking i was wrong.
Thanks for this clarification. 
If you have time please can you tell me one more details, in this [https://github.com/davidsandberg/facenet/issues/152](issue) you mentioned that for 51K classes your machine computed it in 43 hours. On my VM on Azure with K80 GPU, with same parameters and 55K classes of MS-Celeb set it will takes ~7 days. 
Do you have any advice?
Thanks

**Comment by davidsandberg** [comment]
Sorry, no good ideas about increasing training speed. Making sure you are using the latest cuda version is probably my best advice.

**Comment by aleksandar** [comment]
Ok, thanks a lot.
